src/kern/linux/helpers.cc

```diff
@@ -37,7 +37,22 @@
 
 #include "kern/linux/helpers.hh"
 
+/* The following pragmas are to fix a bug in GCC \w the CPP standard library,
+ * outlined here: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105562.
+ * This ignores the 'maybe-unitialized' warning when importing regex.
+ */
+
+// save diagnostic state
+#pragma GCC diagnostic push
+// ignore 'maybe-uniitialized' warnings
+#if !defined(__has_warning) || __has_warning("-Wmaybe-uninitialized")
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#endif
+// include regex
 #include <regex>
```

Comment:
(nit) The compile only failed in gcc 12.1.0 version. The gcc 12.3.0 and 13.2.0 can work without the warning ignore.

Reply:
Thanks! I've set this to only disable this diagnostic when it's GCC 12.1.0.